design-documents/graph-ql/coverage/returns.graphqls

```diff
@@ -0,0 +1,197 @@
+}
+
+enum CustomAttributesPageEnum {
+    RETURN_ITEM_EDIT_FORM
+    RETURN_ITEMS_LISTING
+    # See https://github.com/magento/architecture/blob/master/design-documents/graph-ql/framework/attributes-metadata.md
+    # PRODUCTS_COMPARE
+    # PRODUCTS_LISTING
+    # ADVANCED_CATALOG_SEARCH
+}
+
+type Customer {
+    returns(
+        pageSize: Int = 20 @doc(description: "Specifies the maximum number of results to return at once. Defaults to 20."),
+        currentPage: Int = 1 @doc(description: "Specifies which page of results to return. The default value is 1."),
+    ): Returns @doc(description: "Information about the customer returns.")
+    return(id: ID!): Return @doc(description: "Get customer return details by its ID.")
+}
+
+type CustomerOrder {
+    returns(
+        pageSize: Int = 20 @doc(description: "Specifies the maximum number of results to return at once. Defaults to 20."),
+        currentPage: Int = 1 @doc(description: "Specifies which page of results to return. The default value is 1."),
+    ): Returns @doc(description: "Information about the order returns.")
+    items_eligible_for_return: [OrderItemInterface] @doc(description: "A list of order items eligible for return.")
+}
+
+type Returns {
+    items: [Return] @doc(description: "List of returns")
+    page_info: SearchResultPageInfo @doc(description: "Pagination metadata")
+    total_count: Int @doc(description: "Total count of customer returns")
+}
+
+type Return @doc(description: "Customer return") {
+    id: ID! @doc(description: "Return number")
+    order_id: String! @doc(description: "ID of the order based on which the return was created.")
+    creation_date: String! @doc(description: "The date when the return was requested.")
+    customer_email: String! @doc(description: "Email of the customer who created the return.")
+    customer_name: String @doc(description: "The name of the person returning the item")
+    status: ReturnStatus @doc(description: "Return status.")
+    shipping: ReturnShipping @doc(description: "Shipping information for the return.")
+    comments: [ReturnComment] @doc(description: "A list of comments posted for the return.")
+    items: [ReturnItem] @doc(description: "A list of items being returned.")
+    available_shipping_carriers: [ReturnShippingCarrier] @doc(description: "A list of shipping carriers available for returns.")
+}
+
+type ReturnItem {
+    id: ID!
+    product: ProductInterface!
+    custom_attributes: [CustomAttribute]
+    request_quantity: Float!
+    quantity: Float!
+    status: ReturnItemStatus!
+}
+
+# See https://github.com/magento/architecture/blob/master/design-documents/graph-ql/custom-attributes-container.md
+type CustomAttribute {
+    id: ID!
+    label: String!
+    value: String! @doc(description: "JSON encoded value of the attribute.")
+}
+
+type ReturnComment {
+    id: ID! @doc(description: "Comment ID.")
+    created_by: String! @doc(description: "The name of the user who created posted the comment.")
```

Comment:
Separate firstname/lastname fields tend to be helpful for the client.

Sometimes you want to abbreviate one of the names to accommodate small screen sizes (_A. Paliarush_ vs _Alex Paliarush_)

Reply:
Here's what I did for B2B negotiable quotes

https://github.com/magento/architecture/pull/399/files#diff-4d6fb9dda54abab9b58373010823458bR326-R332

We can unify approaches

Reply:
@nrkapoor would you like us to extend current schema to have separate First name and Last name for comment author?

Reply:
@paliarush yes. that would make it cleaner.

Reply:
Changed to
```
    author_firstname: String! @doc(description: "First name of the user who posted the comment.")
    author_lastname: String! @doc(description: "Last name of the user who posted the comment.")
```